plugins/aggregators/tdigestagg/aggregation.go

```diff
@@ -0,0 +1,172 @@
+//Copyright (c) 2019, Groupon, Inc.
+//All rights reserved.
+//
+//Redistribution and use in source and binary forms, with or without
+//modification, are permitted provided that the following conditions are
+//met:
+//
+//Redistributions of source code must retain the above copyright notice,
+//this list of conditions and the following disclaimer.
+//
+//Redistributions in binary form must reproduce the above copyright
+//notice, this list of conditions and the following disclaimer in the
+//documentation and/or other materials provided with the distribution.
+//
+//Neither the name of GROUPON nor the names of its contributors may be
+//used to endorse or promote products derived from this software without
+//specific prior written permission.
+//
+//THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+//TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+//PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+//HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+//SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+//TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+package tdigestagg
+
+import (
+	"fmt"
+	"github.com/influxdata/tdigest"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/aggregators/tdigestagg/constants"
+	"strconv"
+	"strings"
+	"time"
+)
+
+var weight = 1.
+
+type Aggregation interface {
+	histogram() *tdigest.TDigest
+	tags() map[string]string
+	basicName() string
+	sum() float64
+	time() time.Time
+
+	addValue(floatValue float64)
+	emit(acc telegraf.Accumulator)
+}
+
+type AggregationData struct {
+	_histogram *tdigest.TDigest
+	_tags      map[string]string
+	_basicName string
+	_sum       float64
+	_time      time.Time
```

Comment:
This isn't very idiomatic for Go, name these without the leading underscore and export the function names in the interface.